backends/backend-sdl/src/arc/backend/sdl/jni/SDL.java

```diff
@@ -336,26 +337,47 @@ protected Throwable loadFile(String sourcePath, String sourceCrc, File extracted
                     data[3] = e.window.data2;
                     break;
                 case SDL_MOUSEMOTION:
+                    if(e.motion.which == SDL_TOUCH_MOUSEID) break;
                     data[0] = 2;
                     data[1] = e.motion.x;
-                    data[2] = e.motion.y;
+                    data[2] = (height - e.motion.y);
                     break;
                 case SDL_MOUSEBUTTONDOWN:
                 case SDL_MOUSEBUTTONUP:
+                    if(e.button.which == SDL_TOUCH_MOUSEID) break;
                     data[0] = 3;
                     data[1] = (e.type == SDL_MOUSEBUTTONDOWN);
                     data[2] = e.button.x;
-                    data[3] = e.button.y;
+                    data[3] = (height - e.button.y);
                     data[4] = e.button.button;
                     break;
                 case SDL_MOUSEWHEEL:
+                    if(e.wheel.which == SDL_TOUCH_MOUSEID) break;
                     data[0] = 4;
                     data[1] = e.wheel.x;
                     data[2] = e.wheel.y;
                     break;
+                case SDL_FINGERMOTION:
+                case SDL_FINGERDOWN:
+                case SDL_FINGERUP:
+                    data[0] = 5;
+                    // everything relating to touch (sans gestures) in SDL2 are under one data field,
+                    // this mirrors that rather than splitting them apart.
+                    // 0 - finger motion, 1 - finger down, 2 - finger up
+                    if(e.type == SDL_FINGERMOTION){ data[1] = 0; }
```

Comment:
This may be C code, but please keep it compliant with the style guide, e.g. `}else if(){`, no same-line if-statements, etc.

Reply:
(remedied)